### PR TITLE
feat: #WB2-1592, add visible to revision creation

### PR DESCRIPTION
**Pull request by david-cc**
## Describe your changes

Add isVisible property to page revision information and add some checks about updating a page and creating a revision based on specs : https://edifice-community.atlassian.net/wiki/spaces/ODE/pages/3830874113/Wiki+et+cours+2024+-+Documentation+fonctionnelle#G%C3%A9n%C3%A9rer-une-version

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: backend/src/main/java/net/atos/entng/wiki/controllers/WikiController.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import java.util.function.Function;
 
 import fr.wseduc.webutils.I18n;
+import io.netty.util.internal.StringUtil;
```
Comment (juniorode) [inline backend/src/main/java/net/atos/entng/wiki/controllers/WikiController.java:27]: Il me semble que l'import est inutilisé.

File: backend/src/main/java/net/atos/entng/wiki/controllers/WikiController.java [hunk 2]
```diff
@@ -355,15 +362,47 @@ public void updatePage(final HttpServerRequest request) {
 					final String idWiki = request.params().get("id");
 					page.put("_id", request.params().get("idpage"));
 
-					wikiService.updatePage(user, idWiki, page, request, result -> {
-						if (result.isRight()) {
-							createRevision(idWiki, page.getString("_id"), user, page.getString("title")
-									, page.getString("content"));
-							renderJson(request, page);
+					// get existing page to compare page values
```
Comment (juniorode) [inline backend/src/main/java/net/atos/entng/wiki/controllers/WikiController.java:365]: OK pour le traitement ci-dessous mais je pense que pour bien séparer les couches (controller, services, DAO) il faudrait mettre tout ça dans la méthode `updatePage`.

**Review by juniorode — CHANGES_REQUESTED** [review]
c.f. les commentaires

File: backend/src/main/java/net/atos/entng/wiki/controllers/WikiController.java (hunk 1 above)
Comment (david-cc) [reply backend/src/main/java/net/atos/entng/wiki/controllers/WikiController.java:27]: yes en effet, j'enlève l'import

File: backend/src/main/java/net/atos/entng/wiki/controllers/WikiController.java (hunk 2 above)
Comment (david-cc) [reply backend/src/main/java/net/atos/entng/wiki/controllers/WikiController.java:365]: @juniorode il faudrait mettre tout le traitement dans la méthode `updatePage` du Service c'est ça ?

**Review by juniorode — APPROVED** [review]

**Pull request merged**